apps/src/templates/projects/ProjectsGallery.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useCallback, useMemo, useState} from 'react';
+
+import SegmentedButtons from '@cdo/apps/componentLibrary/segmentedButtons';
+import LibraryTable from '@cdo/apps/templates/projects/LibraryTable';
+import PersonalProjectsTable from '@cdo/apps/templates/projects/PersonalProjectsTable';
+import PublicGallery from '@cdo/apps/templates/projects/PublicGallery';
+import i18n from '@cdo/locale';
+
+import {Galleries} from './projectConstants';
+import {selectGallery} from './projectsRedux';
+
+import moduleStyles from './project-gallery.module.scss';
+
+interface ProjectsGalleryProps {
+  limitedGallery: boolean;
+}
+
+const ProjectsGallery: React.FunctionComponent<ProjectsGalleryProps> = ({
+  limitedGallery,
+}) => {
+  const [selectedTab, setSelectedTab] = useState(Galleries.PRIVATE);
+
+  const galleryTabs = useMemo(() => {
+    const tabs = [
+      {
+        value: Galleries.PRIVATE,
+        label: i18n.myProjects(),
+        url: '/projects',
+        tabContent: <PersonalProjectsTable />,
+      },
+      {
+        value: Galleries.LIBRARIES,
+        label: i18n.myLibraries(),
+        url: '/projects/libraries',
+        tabContent: <LibraryTable />,
+      },
+      {
+        value: Galleries.PUBLIC,
+        label: i18n.featuredProjects(),
+        url: '/projects/public',
+        tabContent: <PublicGallery limitedGallery={limitedGallery} />,
+      },
+    ];
+    return tabs;
+  }, [limitedGallery]);
+
+  const handleOnChange = useCallback(
+    (value: string) => {
+      setSelectedTab(value);
+      selectGallery(value);
```

Comment:
why don't we use the redux value `selectedGallery` rather than the state value?

Reply:
Can I do that without converting the projects redux to typescript? I got muddled with connect v selectors and didn't want to have to define the full type of projects state. But maybe I'm making it harder than it needs to be...?

Reply:
You can manually type it as part of the call to useSelector, something like:
```
useSelector((state: {projects: {selectedGallery: string}) => state.projects.selectedGallery)
```
You only need to define the part of the state type you are using.

Reply:
aha! Thank you!